mantidproject/mantid
# Loader and reduction for Lagrange at ILL

**Description of work.**
This PR introduces a loader and a reduction algorithm for Lagrange data.
Lagrange data being a instrument scan stored in an ASCII file, it was considered easier to manage the loading side directly in the Python algorithm. Due to this, it is not possible to load the data through a general loader (but loading non-reduced data is not very informative, and could be done by running the reduction algorithm without any reduction step).

**To test:**


Run ILLLagrange with

    SampleRuns: 012882:012884,
    ContainerRuns: 012869:012871,
    CorrectionFile: correction-water-cu220-2020.txt

and check the 1D plot make  sense. Also check that an invisible workspace group with intermediate reduction states has been created.

Fixes #34746 .





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
